### PR TITLE
Fixes issue with compose accessibility labels when mergeDescendants is true

### PR DESCRIPTION
**Pull request by adamalyyan**
This fixes an issue where the labels generated when a root node has mergeDescendants to true whereby the full label was not being rendered. This happened when the root had multiple levels of children that didn't have mergeDescendants to true meaning that they should have been appended to the label for the root.

For example:
```
Column(onClickLabel = "Click Label" {}) {
   Column(mergeDescendants=true) {
         Text("Test")
         Text("Test 2")
    }
   Column {
         Text("Test")
         Text("Test 2")
    }
}
```

Previous this would output:
"Test, Test 2"
"Test, Test 2"

But it should output (and now does):
"Test, Test 2, Click Label"
"Test, Test 2"

which matches how Talkback reads out the labels.

An additional benefit of this change is it includes the `Role` in the label for Compose views to match what TalkBack announces and helps avoid cases where "button" is included in the label preventing Talkback from saying "Cancel Button, Button". For example:
```
Button(onClickLabel="Cancel Button")
```

Previously this would output:
"Cancel Button"

But talkback would read this as "Cancel Button, Button" so now the label will show this (and allow the developer to update the label):
"Cancel Button, Button"

### CODE REVIEW

**Review by geoff-powell — APPROVED** [review]
got it this looks good to me, thanks for the in-depth explanation of the issue!

File: paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt [hunk 1]
```diff
@@ -135,6 +128,17 @@ class AccessibilityRenderExtension : RenderExtension {
...
+private fun SemanticsNode.findAllUnmergedNodes(): List<SemanticsNode> {
+  val nodes = mutableListOf<SemanticsNode>()
+  children
+    .filter { !it.config.isMergingSemanticsOfDescendants }
+    .forEach {
+      nodes.addAll(it.findAllUnmergedNodes())
+    }
+  nodes.add(this)
+  return nodes
+}
```
Comment (jrodbx) [inline paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt:140]: using buildList here, should clean this up a bit.

File: paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt [hunk 2]
```diff
@@ -97,22 +97,15 @@ class AccessibilityRenderExtension : RenderExtension {
...
-          // Ignore nodes that merge descendants as they will have their own merged node
-          node.children.filter {
-            !it.config.isMergingSemanticsOfDescendants
-          }.mapNotNull { child ->
-            child.accessibilityText()
-          }.joinToString(", ").ifEmpty { null }
-        }
+        node.findAllUnmergedNodes().mapNotNull {
+          it.accessibilityText()
+        }.joinToString(", ").ifEmpty { null }
```
Comment (jrodbx) [inline paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt:105]:
```suggestion
        node.findAllUnmergedNodes()
          .mapNotNull { it.accessibilityText() }
          .joinToString(", ")
          .ifEmpty { null }
```

**Review by jrodbx — APPROVED** [review]

File: paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt [hunk 3]
```diff
@@ -135,6 +129,17 @@ class AccessibilityRenderExtension : RenderExtension {
   }
 }
 
+private fun SemanticsNode.findAllUnmergedNodes(): List<SemanticsNode> {
+  return buildList {
+    children
+      .filter { !it.config.isMergingSemanticsOfDescendants }
+      .forEach {
+        addAll(it.findAllUnmergedNodes())
+      }
```
Comment (jrodbx) [inline paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt:138]:
```
addAll(
  children
      .filter { !it.config.isMergingSemanticsOfDescendants }
      .map { it.findAllUnmergedNodes() }
)
```

**Review by jrodbx — APPROVED** [review]

File: paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt (hunk 3 above)
Comment (adamalyyan) [reply paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt:138]: Oh, very cool!

**Comment by jrodbx** [comment]
Heads up @adamalyyan, I think the snapshot for the composeA11y test needs updating!

**Comment by jrodbx** [comment]
?

**Comment by jrodbx** [comment]
Not sure why this got auto-closed....?

**Comment by jrodbx** [comment]
Trying to re-open...but I think since:
* we use a default squash + merge policy AND
* this PR was rebased on my PR AND
* my PR was just merged into master, which auto-deletes the remote branch

that resulted in this PR being auto-closed?  sigh.

**Comment by jrodbx** [comment]
Trying again

**Comment by jrodbx** [comment]
Ok, yea, this PR is hosed.  my bad, @adamalyyan mind reopening in another PR?  🤦

**Comment by adamalyyan** [comment]
> Ok, yea, this PR is hosed. my bad, @adamalyyan mind reopening in another PR? 🤦

All good! Here it is: https://github.com/cashapp/paparazzi/pull/933

**Pull request closed without merging**